src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -189,3 +193,27 @@ trait IterablePolyTransforms[+A, +C[A]] extends Any {
   def zip[B](xs: IterableOnce[B]): C[(A @uncheckedVariance, B)] = fromIterable(View.Zip(coll, xs))
   // sound bcs of VarianceNote
 }
+
+/** Transforms over iterables that can return collections of different element types for which an
+  * implicit evidence is required.
+  */
+trait ConstrainedIterablePolyTransforms[+A, +C[A], +CC[X] <: C[X], Ev[A]] extends ConstrainedFromIterable[CC, Ev] with IterablePolyTransforms[A, C] {
```

Comment:
I already removed a bunch of unnecessary names. I must have overlooked these ones.